zproject/backends.py

```diff
@@ -2150,6 +2167,8 @@ class AzureADAuthBackend(SocialAuthMixin, AzureADOAuth2):
     auth_backend_name = "AzureAD"
     display_icon = staticfiles_storage.url("images/authentication_backends/azuread-icon.png")
 
+    minimum_realm_plan_type = Realm.PLAN_TYPE_STANDARD
```

Comment:
Here we can mark backends with a required `plan_type`. The rest of the changes in the commit are about enforcing this correctly and plumbing the info the frontend

Reply:
We might want to just have this be `allowed_could_plans = [...]`; I don't like doing comparisons of plan IDs just because such code is very fragile when refactored.